src/producer.c

```diff
@@ -2,15 +2,36 @@
 
 /**
  *
- * Producer thread will read from the file and write data to 
+ * Producer thread will read from the file and write data to
  * the end of the shared queue
  */
 void *producer(void *arg){
-    
+
+    printf("producer\n");
+
     //TODO: open the file and read its content line by line
-    //Send data to the shared queue
+
+    FILE * fd_in = (FILE*)arg; //  file we are reading from; will be passed in
+    ssize_t nread;
+    char buffer[chunkSize]; // chunk size for now
+    size_t len = chunkSize;
+
+    // read until EOF
+    while (nread = getLineFromFile(fd_in, buffer, len) != -1) {
+      //Send data to the shared queue
+      struct Packet* p = (struct Packet*) malloc(sizeof(struct Packet));
+      p->line = buffer;
+
+      struct Node* n = (struct Node*)malloc(sizeof(struct Node));
+      n->next = NULL;
+      n->packet = p;
+
+      //enqueue(q, n);  q is not recognized at the moment for some reason.
```

Comment:
you have to declare q just like it is declared in utils.h but without the extern. 

`struct Queue* q;`

I think that is all you have to do and then you can enqueue and such. Not totally sure. Will be interesting to test.

Reply:
Also while adding to the queue there should be some mutual exclusion going on because the consumer is going to need to modify the queue as well.

There should also be some signalling mechanism to let consumers know items have been added. try utilizing the global semaphores (not externs)  `mutex` and `staged`.

You may have to merge changes from main into your branch in order to use these.